src/Microsoft.PowerApps.TestEngine/PowerApps/PowerAppFunctions.cs

```diff
@@ -275,8 +276,13 @@ public async Task<bool> SetPropertyDateAsync(ItemPath itemPath, DateValue value)
                 var propertyNameString = JsonConvert.SerializeObject(itemPath.PropertyName);
                 var recordValue = value.GetConvertedValue(null);
 
-                // Date.parse() parses the date to unix timestamp
-                var expression = $"PowerAppsTestEngine.setPropertyValue({itemPathString},{{{propertyNameString}:Date.parse(\"{recordValue}\")}})";
+                // Date.parse() parses the date to unix timestamp.
+                // SetProperty value expected from user is Date(yyyy, MM, dd)
+                // Setting the time value to T00:00.000Z to maintain uniformity across timezones.
+                // This value explicitly specifies the UTC timezone. 
+                // This helps in fetching the date value as it is in any timezone locally without considering one-off in day value.
+                var dt = $"Date.parse(\"{recordValue.Date.ToString("yyyy-MM-dd")}{UTCTimeValue}\")";
+                var expression = $"PowerAppsTestEngine.setPropertyValue({itemPathString},{{{propertyNameString}:{dt}}})";
```

Comment:
Are there other places where this might be called other than the DatePicker which might expect the time to be included?

Reply:
No, this is specific and only to SetProperty with date value.

Reply:
Sorry, I don't think I was clear. Might it be possible that someone is calling SetProperty on a control that isn't a DatePicker? For example, is there a control which might be expecting the Time to be included & not replaced with 0000Z?